src/System.Text.Json/tests/Serialization/CustomConverterTests.Dictionary.cs

```diff
@@ -86,5 +87,119 @@ public static void CustomDictionaryConverter()
                 Assert.Equal(json, jsonSerialized);
             }
         }
+
+        [Fact]
+        public static void CustomDictionaryConverterContravariant()
+        {
+            const string json = @"{""Key1"":1,""Key2"":2}";
+
+            var options = new JsonSerializerOptions();
+            options.Converters.Add(new IDictionaryConverter(10));
+
+            Dictionary<string, long> dictionary = JsonSerializer.Deserialize<Dictionary<string, long>>(json, options);
+            Assert.Equal(11, dictionary["Key1"]);
+            Assert.Equal(12, dictionary["Key2"]);
+
+            string jsonSerialized = JsonSerializer.Serialize(dictionary, options);
+            Assert.Equal(json, jsonSerialized);
```

Comment:
This may just be personal style preference, but note that you can just write:
```C#
Assert.Equal(json, JsonSerializer.Serialize(dictionary, options));
```